app/controllers/groups_controller.rb

```diff
@@ -198,13 +198,16 @@ def assign_student_and_next
       end
       # if the user has typed in the whole name without select, or if they typed a name different from the select s_id
       if student.nil? || ("#{student.first_name} #{student.last_name}") != params[:names]
-        student = current_course.students.where(
+        student = current_course.students.joins(:end_user).where(
           'lower(CONCAT(first_name, \' \', last_name)) like ? OR lower(CONCAT(last_name, \' \', first_name)) like ?',
           params[:names].downcase, params[:names].downcase
         ).first
       end
+      if student.nil?
+        flash_message(:error, t('exam_templates.assign_scans.student_not_found', name: params[:names]))
+      end
```

Comment:
Add a `head :bad_request` (or the appropriate status) and `return` here. This will prevent the browser from going to the next scan (but the flash message will still appear).